src/main.rs

```diff
@@ -185,8 +176,39 @@ async fn main() -> Result<()> {
         eprintln!("Test exited with code {}, so aborting test.", status);
         exit(status);
     }
+    get_kernel_metrics(&mut metrics);
+    get_statsd_metrics(&mut metrics, statsd_buf.read().await.clone())?;
+    statsd_buf.write().await.clear();
+    Ok(())
+}
+
+#[async_std::main]
+async fn main() -> Result<()> {
+    let config_file = env::args().nth(1).expect("missing file argument");
+    let config = get_config(&config_file)?;
+    if let Some(setup) = &config.setup {
+        if env::var("SIRUN_SKIP_SETUP").is_err() {
+            run_setup(&setup, &config_file, &config.env).await?;
+        }
+    }
+
+    let statsd_started = Arc::new(Barrier::new(2));
+    let statsd_buf = Arc::new(RwLock::new(String::new()));
+    spawn(statsd_listener(statsd_started.clone(), statsd_buf.clone()));
+    statsd_started.wait().await; // waits for socket to be listening
 
     let mut metrics: HashMap<String, MetricValue> = HashMap::new();
+    if config.iterations == 1 {
+        run_iteration(&config, &mut metrics, statsd_buf.clone()).await?;
+    } else {
+        let mut iterations = Vec::new();
+        for _ in 0..config.iterations {
+            let mut iteration_metrics = HashMap::new();
+            run_iteration(&config, &mut iteration_metrics, statsd_buf.clone()).await?;
```

Comment:
I am likely over-engineering, but should we inject some SIRUN env variables into the command?

`SIRUN_ITERATION=5` ?

(non-blocking comment, random thought)

Reply:
Is there a use case where you'd want to do something different for each iteration? The results are retrieved serially, so if it's only to include in output, you already have the ordering of the array.

Reply:
yeah, I suppose I saw `echo "this is something"` and thought... hey, `echo "iteration: ${SIRUN_ITTERATION}"` but I suppose doesn't really add much value other than that 😆

Reply:
Yeah I think if you want to switch based on it, you probably actually want a variant.

Running all variants seems like a desirable feature, but out of scope for this PR.